### PR TITLE
test: improve pkg/gitinterface coverage from 69.4% to 76.2%

### PR DESCRIPTION
**Pull request by dawgdevv**
## Description

Improves test coverage for the pkg/gitinterface package from 69.4% to 76.2%, an increase of 6.8 percentage points.

This PR adds focused test cases for error paths and edge cases across 9 files.
All tests follow the existing patterns in the codebase (t.Run subtests, CreateTestGitRepository, require.Nil/assert.Equal).

| File                 | What was added                                                                                                      |
|----------------------|----------------------------------------------------------------------------------------------------------------------|
| references_test.go   | TestAbsoluteReference — tests for HEAD, branches, tags, custom refs, remote refs, and ErrReferenceNotFound          |
| tag_test.go          | TestEnsureIsTag + error-path t.Run for GetTagTarget and verifyTagSignature (unknown signing method)                 |
| commit_test.go       | TestEnsureIsCommit + error-path t.Run for GetCommitMessage, GetCommitTreeID, GetCommitParentIDs, KnowsCommit, GetCommonAncestor, and verifyCommitSignature |
| object_test.go       | Error-path t.Run for GetObjectType and GetObjectSize with non-existent objects                                      |
| blob_test.go         | Error-path t.Run for ReadBlob with a non-blob object                                                                 |
| status_test.go       | TestStatusError, TestNewStatusCodeFromByte, TestStatusCodeString, TestFileStatusUntracked                            |
| sync_test.go         | TestWithFetchDepth, TestCreateRemote, and error-path t.Run for push/fetch with non-existent remotes and invalid refs |
| config_test.go       | Error-path t.Run for SetGitConfig with invalid key                                                                   |
| utils_test.go        | TestResetDueToError — successful reset and invalid ref name                                                          |

Related to #1244

## AI Usage

- [ ] I **did not** use generative AI at all in making the content of this pull
  request.
- [x] I **did** use generative AI in some form in making the content of this
  pull request. I have described my use of AI below.

I used an AI coding assistant to help identify gaps in test coverage by analyzing the coverage report and suggesting which error paths were uncovered. I wrote all test logic manually based on the existing patterns in the `pkg/gitinterface/*_test.go` files. I used the AI assistant to verify that my test structure matched the project's conventions (using `t.Run`, `require.Nil`, `assert.ErrorIs`, etc.). All test assertions, error message substrings, and test data were manually reviewed by me to ensure they correctly test the intended functionality. I made significant manual edits to every AI-suggested snippet before committing.

## Contributor Checklist

- [x] I **have manually reviewed all content** submitted to gittuf in this pull
  request.
- [x] I fully understand the content I am submitting.
- [x] The changes introduced are documented and have tests included if
  applicable.
- [x] My changes do not infringe on copyright/trademarks/etc.
- [x] All commits in this pull request include a [DCO
  Signoff](https://wiki.linuxfoundation.org/dco).
- [x] By submitting this pull request, I agree to follow the gittuf [Code of
  Conduct](https://github.com/gittuf/community/blob/main/CODE-OF-CONDUCT.md).

### CODE REVIEW

File: pkg/gitinterface/blob_test.go [hunk 1]
```diff
@@ -34,6 +34,14 @@ func TestRepositoryReadBlob(t *testing.T) {
 		_, err := repo.ReadBlob(ZeroHash)
 		assert.NotNil(t, err)
 	})
+
+	t.Run("read non-blob object", func(t *testing.T) {
+		treeBuilder := NewTreeBuilder(repo)
+		treeID, err := treeBuilder.WriteTreeFromEntries(nil)
+		require.Nil(t, err)
+		_, err = repo.ReadBlob(treeID)
```
Comment (patzielinski) [inline pkg/gitinterface/blob_test.go:43]:
```suggestion

		_, err = repo.ReadBlob(treeID)
```

File: pkg/gitinterface/blob_test.go [hunk 2]
```diff
@@ -34,6 +34,14 @@ func TestRepositoryReadBlob(t *testing.T) {
...
+		assert.NotNil(t, err)
```
Comment (patzielinski) [inline pkg/gitinterface/blob_test.go]: Please use `assert.ErrorIs` if there is a fixed error the function returns, or `assert.ErrorContains` and match on part of the error if the error is variable or not defined (e.g. `Err...`)

Please make this change throughout the PR.

File: pkg/gitinterface/commit_test.go [hunk 3]
```diff
@@ -247,6 +249,12 @@ func TestRepositoryVerifyCommit(t *testing.T) {
 		err = repo.verifyCommitSignature(context.Background(), gitsignSignedCommitID, sshKey)
 		assert.ErrorIs(t, err, ErrIncorrectVerificationKey)
 	})
+
+	t.Run("unknown signing method", func(t *testing.T) {
+		unknownKey := &signerverifier.SSLibKey{KeyType: "unknown"}
+		err = repo.verifyCommitSignature(context.Background(), sshSignedCommitID, unknownKey)
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go]:
```suggestion
		err = repo.verifyCommitSignature(t.Context(), sshSignedCommitID, unknownKey)
```

File: pkg/gitinterface/commit_test.go [hunk 4]
```diff
@@ -304,6 +312,20 @@ func TestKnowsCommit(t *testing.T) {
 		knows, _ := repo.KnowsCommit(unknownCommitID, firstCommitID)
 		assert.False(t, knows)
 	})
+
+	t.Run("non-commit object as first arg", func(t *testing.T) {
+		blobID, err := repo.WriteBlob([]byte("test"))
+		require.Nil(t, err)
+		_, err = repo.KnowsCommit(blobID, firstCommitID)
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go:320]:
```suggestion

				_, err = repo.KnowsCommit(blobID, firstCommitID)
```

File: pkg/gitinterface/commit_test.go [hunk 5]
```diff
@@ -304,6 +312,20 @@ func TestKnowsCommit(t *testing.T) {
...
+		assert.NotNil(t, err)
+	})
+
+	t.Run("non-commit object as second arg", func(t *testing.T) {
+		blobID, err := repo.WriteBlob([]byte("test"))
+		require.Nil(t, err)
+		_, err = repo.KnowsCommit(firstCommitID, blobID)
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go:328]:
```suggestion

		_, err = repo.KnowsCommit(firstCommitID, blobID)
```

File: pkg/gitinterface/commit_test.go [hunk 6]
```diff
@@ -457,6 +479,13 @@ func TestRepositoryGetCommitMessage(t *testing.T) {
 	commitMessage, err := repo.GetCommitMessage(commit)
 	assert.Nil(t, err)
 	assert.Equal(t, message, commitMessage)
+
+	t.Run("non-commit object", func(t *testing.T) {
+		blobID, err := repo.WriteBlob([]byte("test"))
+		require.Nil(t, err)
+		_, err = repo.GetCommitMessage(blobID)
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go:489]:
```suggestion

		_, err = repo.GetCommitMessage(blobID)
```

File: pkg/gitinterface/commit_test.go [hunk 7]
```diff
@@ -501,6 +530,13 @@ func TestGetCommitTreeID(t *testing.T) {
 	secondCommitTreeID, err := repo.GetCommitTreeID(secondCommitID)
 	assert.Nil(t, err)
 	assert.Equal(t, treeWithContentsID, secondCommitTreeID)
+
+	t.Run("non-commit object", func(t *testing.T) {
+		blobID, err := repo.WriteBlob([]byte("test"))
+		require.Nil(t, err)
+		_, err = repo.GetCommitTreeID(blobID)
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go:541]:
```suggestion
		
		_, err = repo.GetCommitTreeID(blobID)
```

File: pkg/gitinterface/commit_test.go [hunk 8]
```diff
@@ -537,6 +573,13 @@ func TestGetCommitParentIDs(t *testing.T) {
 	secondCommitParentIDs, err := repo.GetCommitParentIDs(secondCommitID)
 	assert.Nil(t, err)
 	assert.Equal(t, []Hash{initialCommitID}, secondCommitParentIDs)
+
+	t.Run("non-commit object", func(t *testing.T) {
+		blobID, err := repo.WriteBlob([]byte("test"))
+		require.Nil(t, err)
+		_, err = repo.GetCommitParentIDs(blobID)
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go]:
```suggestion
		
		_, err = repo.GetCommitParentIDs(blobID)
```

File: pkg/gitinterface/commit_test.go [hunk 9]
```diff
@@ -577,4 +620,48 @@ func TestGetCommonAncestor(t *testing.T) {
 
 	_, err = repo.GetCommonAncestor(commitDisconnected, commitA)
 	assert.NotNil(t, err)
+
+	t.Run("non-commit object as first arg", func(t *testing.T) {
+		blobID, err := repo.WriteBlob([]byte("test"))
+		require.Nil(t, err)
+		_, err = repo.GetCommonAncestor(blobID, commitA)
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go:630]:
```suggestion
		
		_, err = repo.GetCommonAncestor(blobID, commitA)
```

File: pkg/gitinterface/commit_test.go [hunk 10]
```diff
@@ -577,4 +620,48 @@ func TestGetCommonAncestor(t *testing.T) {
...
+		assert.NotNil(t, err)
+	})
+
+	t.Run("non-commit object as second arg", func(t *testing.T) {
+		blobID, err := repo.WriteBlob([]byte("test"))
+		require.Nil(t, err)
+		_, err = repo.GetCommonAncestor(commitA, blobID)
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go:638]:
```suggestion
		
		_, err = repo.GetCommonAncestor(commitA, blobID)
```

File: pkg/gitinterface/references_test.go [hunk 11]
```diff
@@ -389,3 +390,66 @@ func TestRemoteReferenceName(t *testing.T) {
...
+
+	err = repo.SetReference("refs/custom/myref", commitID)
+	require.Nil(t, err)
+
+	err = repo.SetReference("refs/remotes/origin/main", commitID)
+	require.Nil(t, err)
+
+	err = repo.SetSymbolicReference("refs/remotes/origin/HEAD", "refs/remotes/origin/main")
+	require.Nil(t, err)
+	t.Run("symbolic ref HEAD", func(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/references_test.go:417]:
```suggestion
	
	t.Run("symbolic ref HEAD", func(t *testing.T) {
```

File: pkg/gitinterface/object_test.go [hunk 12]
```diff
@@ -98,6 +98,11 @@ func TestGetObjectType(t *testing.T) {
 	objType, err = repo.GetObjectType(tagID)
 	assert.Nil(t, err)
 	assert.Equal(t, TagObjectType, objType)
+
+	t.Run("non-existent object", func(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/object_test.go]: Please change this from a `t.Run` to inline.

File: pkg/gitinterface/object_test.go [hunk 13]
```diff
@@ -110,4 +115,9 @@ func TestGetObjectSize(t *testing.T) {
 	objSize, err := repo.GetObjectSize(blobID)
 	assert.Nil(t, err)
 	assert.Equal(t, uint64(6), objSize)
+
+	t.Run("non-existent object", func(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/object_test.go]: Please change this from a `t.Run` to inline.

File: pkg/gitinterface/references_test.go [hunk 14]
```diff
@@ -389,3 +390,66 @@ func TestRemoteReferenceName(t *testing.T) {
...
+		ref, err := repo.AbsoluteReference("HEAD")
+		assert.Nil(t, err)
+		assert.Equal(t, "refs/heads/main", ref)
+	})
```
Comment (patzielinski) [inline pkg/gitinterface/references_test.go:421]:
```suggestion
	})

```

Please also apply for the other `t.Run`s (except for the last one).

File: .gitignore
Comment (patzielinski) [inline .gitignore:1]: Please remove edits to this file.

File: pkg/gitinterface/sync_test.go [hunk 15]
```diff
@@ -153,6 +153,13 @@ func TestPushRefSpecRepository(t *testing.T) {
 		err = localRepo.PushRefSpec(remoteName, []string{refSpecs})
 		assert.Nil(t, err)
 	})
+
+	t.Run("push to non-existent remote", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+		localRepo := CreateTestGitRepository(t, localTmpDir, false)
+		err := localRepo.PushRefSpec("nonexistent", []string{refSpecs})
```
Comment (patzielinski) [inline pkg/gitinterface/sync_test.go:161]:
```suggestion
		
		err := localRepo.PushRefSpec("nonexistent", []string{refSpecs})
```

File: pkg/gitinterface/sync_test.go [hunk 16]
```diff
@@ -294,6 +301,21 @@ func TestPushRepository(t *testing.T) {
 		err = localRepo.Push(remoteName, []string{refName})
 		assert.Nil(t, err)
 	})
+
+	t.Run("push to non-existent remote", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+		localRepo := CreateTestGitRepository(t, localTmpDir, false)
+		err := localRepo.Push("nonexistent", []string{refName})
```
Comment (patzielinski) [inline pkg/gitinterface/sync_test.go:310]:
```suggestion
		
		err := localRepo.Push("nonexistent", []string{refName})
```

File: pkg/gitinterface/sync_test.go [hunk 17]
```diff
@@ -294,6 +301,21 @@ func TestPushRepository(t *testing.T) {
...
+		assert.NotNil(t, err)
+	})
+
+	t.Run("push with invalid ref", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+		localRepo := CreateTestGitRepository(t, localTmpDir, false)
+		err := localRepo.Push("origin", []string{"nonexistent-ref"})
```
Comment (patzielinski) [inline pkg/gitinterface/sync_test.go:318]:
```suggestion
		
		err := localRepo.Push("origin", []string{"nonexistent-ref"})
```

File: pkg/gitinterface/sync_test.go [hunk 18]
```diff
@@ -443,6 +465,13 @@ func TestFetchRefSpecRepository(t *testing.T) {
 		require.Nil(t, err)
 		assert.Equal(t, localRef, newLocalRef)
 	})
+
+	t.Run("fetch from non-existent remote", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+		localRepo := CreateTestGitRepository(t, localTmpDir, true)
+		err := localRepo.FetchRefSpec("nonexistent", []string{refSpecs})
```
Comment (patzielinski) [inline pkg/gitinterface/sync_test.go:475]:
```suggestion
		
		err := localRepo.FetchRefSpec("nonexistent", []string{refSpecs})
```

File: pkg/gitinterface/sync_test.go [hunk 19]
```diff
@@ -591,6 +620,24 @@ func TestFetchRepository(t *testing.T) {
 		require.Nil(t, err)
 		assert.Equal(t, localRef, newLocalRef)
 	})
+
+	t.Run("fetch from non-existent remote", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+		localRepo := CreateTestGitRepository(t, localTmpDir, true)
+		err := localRepo.Fetch("nonexistent", []string{refName}, true)
```
Comment (patzielinski) [inline pkg/gitinterface/sync_test.go:631]:
```suggestion
		
		err := localRepo.Fetch("nonexistent", []string{refName}, true)
```

File: pkg/gitinterface/sync_test.go [hunk 20]
```diff
@@ -591,6 +620,24 @@ func TestFetchRepository(t *testing.T) {
...
+
+	t.Run("fetch with invalid ref", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+		remoteTmpDir := t.TempDir()
+		localRepo := CreateTestGitRepository(t, localTmpDir, true)
+		_ = CreateTestGitRepository(t, remoteTmpDir, false)
+		err := localRepo.CreateRemote(remoteName, remoteTmpDir)
+		require.Nil(t, err)
+		err = localRepo.Fetch(remoteName, []string{"nonexistent-ref"}, true)
+		assert.ErrorIs(t, err, ErrReferenceNotFound)
```
Comment (patzielinski) [inline pkg/gitinterface/sync_test.go:645]:
```suggestion
		localRepo := CreateTestGitRepository(t, localTmpDir, true)
		_ = CreateTestGitRepository(t, remoteTmpDir, false)
		
		err := localRepo.CreateRemote(remoteName, remoteTmpDir)
		require.Nil(t, err)
		
		err = localRepo.Fetch(remoteName, []string{"nonexistent-ref"}, true)
		assert.ErrorIs(t, err, ErrReferenceNotFound)
```

File: pkg/gitinterface/sync_test.go [hunk 21]
```diff
@@ -913,3 +965,21 @@ func TestCloneAndFetchRepository(t *testing.T) {
 		assert.Equal(t, "FETCH_HEAD", dirEntries[0].Name())
 	})
 }
+
+func TestWithFetchDepth(t *testing.T) {
+	options := &FetchOptions{}
+	opt := WithFetchDepth(10)
+	opt(options)
+	assert.Equal(t, 10, options.Depth)
+}
```
Comment (patzielinski) [inline pkg/gitinterface/sync_test.go]:
```suggestion
```

Please remove, we normally don't test options like this.

File: pkg/gitinterface/sync_test.go [hunk 22]
```diff
@@ -913,3 +965,21 @@ func TestCloneAndFetchRepository(t *testing.T) {
...
+
+func TestCreateRemote(t *testing.T) {
+	tmpDir := t.TempDir()
+	repo := CreateTestGitRepository(t, tmpDir, false)
+	err := repo.CreateRemote("origin", tmpDir)
+	assert.Nil(t, err)
+	t.Run("duplicate remote name", func(t *testing.T) {
+		err := repo.CreateRemote("origin", tmpDir)
+		assert.NotNil(t, err)
+	})
```
Comment (patzielinski) [inline pkg/gitinterface/sync_test.go]: This can be inlined.

File: pkg/gitinterface/tag_test.go [hunk 23]
```diff
@@ -92,4 +99,35 @@ func TestRepositoryVerifyTag(t *testing.T) {
...
+
+	t.Run("valid tag", func(t *testing.T) {
+		err := repo.ensureIsTag(tagID)
+		assert.Nil(t, err)
+	})
+
+	t.Run("non-tag object", func(t *testing.T) {
+		err := repo.ensureIsTag(commitID)
+		assert.NotNil(t, err)
+	})
```
Comment (patzielinski) [inline pkg/gitinterface/tag_test.go]: These can be inlined.

File: pkg/gitinterface/utils_test.go [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+	emptyTreeID, err := treeBuilder.WriteTreeFromEntries(nil)
+	require.Nil(t, err)
+	commitID, err := repo.Commit(emptyTreeID, "refs/heads/main", "Initial commit\n", false)
+	require.Nil(t, err)
+	t.Run("successful reset", func(t *testing.T) {
+		cause := assert.AnError
+		err := repo.ResetDueToError(cause, "refs/heads/main", commitID)
+		assert.ErrorIs(t, err, cause)
+	})
+	t.Run("invalid ref name", func(t *testing.T) {
```
Comment (patzielinski) [inline pkg/gitinterface/utils_test.go:29]:
```suggestion
	tmpDir := t.TempDir()
	repo := CreateTestGitRepository(t, tmpDir, false)
	
	treeBuilder := NewTreeBuilder(repo)
	emptyTreeID, err := treeBuilder.WriteTreeFromEntries(nil)
	require.Nil(t, err)
	
	commitID, err := repo.Commit(emptyTreeID, "refs/heads/main", "Initial commit\n", false)
	require.Nil(t, err)
	
	t.Run("successful reset", func(t *testing.T) {
		cause := assert.AnError
		err := repo.ResetDueToError(cause, "refs/heads/main", commitID)
		assert.ErrorIs(t, err, cause)
	})
	
	t.Run("invalid ref name", func(t *testing.T) {
```

File: pkg/gitinterface/utils_test.go [hunk 25]
```diff
@@ -0,0 +1,32 @@
...
+		cause := assert.AnError
+		err := repo.ResetDueToError(cause, "invalid ref with spaces", commitID)
+		assert.NotNil(t, err)
+		assert.Contains(t, err.Error(), "unable to reset")
+		assert.ErrorIs(t, err, cause)
```
Comment (patzielinski) [inline pkg/gitinterface/utils_test.go]: I'd just have an `assert.ErrorContains` here, no?

**Review by patzielinski — CHANGES_REQUESTED** [review]
Please also add a quick description of your use of AI in the PR description, thanks!

File: pkg/gitinterface/utils_test.go (hunk 25 above)
Comment (dawgdevv) [reply pkg/gitinterface/utils_test.go]: I'd like to have both assert.ErrorContains verifies the error message has the right context and assert.ErrorIs verifies the original cause per function contract although i have removed assert.NotNil as it was redundant

**Comment by dawgdevv** [comment]
@patzielinski made specified changes take a look if something still need improvements

File: pkg/gitinterface/commit_test.go (hunk 8 above)
Comment (patzielinski) [reply pkg/gitinterface/commit_test.go]: Bump on this.

File: pkg/gitinterface/commit_test.go [hunk 26]
```diff
@@ -576,5 +623,51 @@ func TestGetCommonAncestor(t *testing.T) {
 	commitDisconnected := repo.commitWithParents(t, emptyTreeID, nil, "Disconnected initial commit\n", false)
 
 	_, err = repo.GetCommonAncestor(commitDisconnected, commitA)
-	assert.NotNil(t, err)
+	assert.ErrorContains(t, err, "merge-base")
```
Comment (patzielinski) [inline pkg/gitinterface/commit_test.go]: Let's leave this line as-is (`assert.NotNil`), it's part of the existing test and the string there isn't too much stronger of an assert.

**Review by patzielinski — COMMENTED** [review]
Looks good, just two nits. Please squash commits down to one and we'll get this merged, thanks!

**Review by patzielinski — APPROVED** [review]
